pkg/controllers/management/template/template_global_catalog.go

```diff
@@ -0,0 +1,74 @@
+package template
+
+import (
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"sort"
+
+	"k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+)
+
+func (tm *RBACTemplateManager) syncForGlobalCatalog(key string, obj *v3.Catalog) error {
+	var templateRuleExists, templateVersionsRuleExists bool
+	// Not going to check here whether obj is nil/deleted or not. Whether global catalog is created, updated or deleted, this method reconciles
+	// the global user role. It gets all templates/templateversions for the global catalogs and updates the user role
+
+	//Get all templates created for global catalogs only and add them via resourceNames to the "user" role
+	r, err := labels.NewRequirement(catalogTypeLabel, selection.Equals, []string{"globalCatalog"})
+	if err != nil {
+		return err
+	}
+
+	userRole, err := tm.globalRoleClient.Get("user", metav1.GetOptions{})
+	userRole = userRole.DeepCopy()
+
+	templates, templateVersions, err := tm.getTemplateAndTemplateVersions(r)
+	sort.Strings(templates)
+	sort.Strings(templateVersions)
+	newTemplateRule := []v1.PolicyRule{
+		{
+			APIGroups:     []string{"management.cattle.io"},
+			Resources:     []string{"templates"},
+			ResourceNames: templates,
+			Verbs:         []string{"get", "list", "watch"},
+		},
+	}
+
+	newTemplateVersionRule := []v1.PolicyRule{
+		{
+			APIGroups:     []string{"management.cattle.io"},
+			Resources:     []string{"templateversions"},
```

Comment:
lets put templateversions and templates in constants, as it's used in a bunch of places